app/models/manageiq/providers/vmware/infra_manager/inventory/collector/property_collector.rb

```diff
@@ -233,75 +44,35 @@ def prop_set
     end
   end
 
-  def folder_to_child_entity
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsFolder', :type => 'Folder', :path => 'childEntity',
-      :selectSet => [
-        RbVmomi::VIM.SelectionSpec(:name => 'tsFolder'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsDcToDsFolder'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsDcToHostFolder'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsDcToNetworkFolder'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsDcToVmFolder'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsCrToHost'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsCrToRp'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsRpToRp'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsRpToVm')
-      ]
-    )
-  end
-
-  def datacenter_to_datastore_folder
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsDcToDsFolder', :type => 'Datacenter', :path => 'datastoreFolder',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsFolder')]
-    )
-  end
-
-  def datacenter_to_host_folder
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsDcToHostFolder', :type => 'Datacenter', :path => 'hostFolder',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsFolder')]
-    )
-  end
-
-  def datacenter_to_network_folder
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsDcToNetworkFolder', :type => 'Datacenter', :path => 'networkFolder',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsFolder')]
-    )
-  end
-
-  def datacenter_to_vm_folder
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsDcToVmFolder', :type => 'Datacenter', :path => 'vmFolder',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsFolder')]
-    )
+  def root_folder_traversal_spec(root)
+    RbVmomi::VIM.ObjectSpec(:obj => root, :selectSet => root_folder_select_set)
   end
 
-  def compute_resource_to_host
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsCrToHost', :type => 'ComputeResource', :path => 'host',
-    )
-  end
-
-  def compute_resource_to_resource_pool
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsCrToRp', :type => 'ComputeResource', :path => 'resourcePool',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsRpToRp')]
+  def root_folder_select_set
+    traversal_spec_from_hash(
+      YAML.load_file(
+        ManageIQ::Providers::Vmware::Engine.root.join(
+          "db/fixtures/traversal_specs/root_folder.yml"
+        )
+      )
     )
   end
 
-  def resource_pool_to_resource_pool
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsRpToRp', :type => 'ResourcePool', :path => 'resourcePool',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsRpToRp')]
-    )
+  def license_manager_traversal_spec(license_manager)
+    RbVmomi::VIM.ObjectSpec(:obj => license_manager)
```

Comment:
Why doesn't this one use traversal_spec_from_hash ?

Reply:
Because this isn't actually a traversal spec, it is just an object in an ObjectSpec without a SelectSpec/TraversalSpec attribute.

I think the issue here is the methods are badly named, before we added the license and extension managers we didn't load anything that didn't have a traversal spec.  I'll fix the names to match the VIM object types that we're actually returning here which should make this cleaner/more obvious.